mama/c_cpp/src/c/datetime.c

```diff
@@ -912,12 +937,24 @@ mamaDateTime_getWithHints(const mamaDateTime     dateTime,
     if (!dateTime || !seconds || !microseconds)
         return MAMA_STATUS_INVALID_ARG;
 
-    *seconds      = mamaDateTimeImpl_getSeconds      (*dateTime);
-    *microseconds = mamaDateTimeImpl_getMicroSeconds (*dateTime);
+    if (mamaDateTimeImpl_getSeconds ((mama_time_t*)dateTime) > UINT32_MAX)
```

Comment:
Or < 0? In fact a small function or macro might help here i've seen this code a lot.